asdf jdk provider

**Is your feature request related to a problem? Please describe.**
Please, add asdf jdk provider

**Describe the solution you'd like**
It would be good to have asdf jdk provider

**Additional context**
The latest jbang (0.102.0) does not work with my jdk, installed by asdf, while the previous one (0.101.0) worked fine with it.

Sure. Here is it:
$ jbang jdk list --show-details
Installed JDKs (<=default):
   11 (11.0.17, current, current, /home/iharh/.asdf/installs/java/corretto-11.0.17.8.1)
   11 (11.0.17, javahome, javahome, /home/iharh/.asdf/installs/java/corretto-11.0.17.8.1)
   11 (11.0.17, path, path, /home/iharh/.asdf/installs/java/corretto-11.0.17.8.1)

I was running simple groovy script:
///usr/bin/env jbang "$0" "$@" ; exit $?
println("Hello")

And the error was:
[jbang] Resolving dependencies...
[jbang]    org.codehaus.groovy:groovy:3.0.9
[jbang] Dependencies resolved
[jbang] Building jar...
[jbang] [ERROR] Error during compile
[jbang] Run with --verbose for more details
groovyc: JAVA_HOME is not defined correctly, can not execute: 11 (11.0.17, current, /home/iharh/.asdf/installs/java/corretto-11.0.17.8.1)/bin/java

--- a/src/main/java/dev/jbang/source/sources/GroovySource.java
+++ b/src/main/java/dev/jbang/source/sources/GroovySource.java
@@ -96,7 +96,7 @@ public class GroovySource extends Source {
 				if (project.getMainSource() instanceof GroovySource) {
 					processBuilder	.environment()
 									.put("JAVA_HOME",
-											JdkManager.getOrInstallJdk(project.getJavaVersion()).toString());
+											JdkManager.getOrInstallJdk(project.getJavaVersion()).getHome().toString());
 					processBuilder.environment().remove("GROOVY_HOME");
 				}
 				super.runCompiler(processBuilder);

